ctk/idp/src/main/kotlin/org/codice/compliance/web/sso/RedirectSSOTest.kt

```diff
@@ -119,7 +117,41 @@ class RedirectSSOTest : StringSpec() {
                     getServiceProvider(IdpSSOResponder::class).getResponseForRedirectRequest(
                             response)
             val samlResponseDom = finalHttpResponse.getBindingVerifier().decodeAndVerify()
-            val samlRequestDom = authnRequest.dom as Node
+
+            // Main goal of this test is to verify the ACS in verifyAssertionConsumerService
+            CoreAuthnRequestProtocolVerifier(authnRequest, samlResponseDom).apply {
+                verify()
+                verifyAssertionConsumerService(finalHttpResponse)
+            }
+            SingleSignOnProfileVerifier(authnRequest, samlResponseDom).apply {
+                verify()
+                verifyBinding(finalHttpResponse)
+            }
+        }
+
+        "Redirect AuthnRequest Using DSA1 Signature Algorithm" {
+            Log.debugWithSupplier { "Redirect AuthnRequest Using DSA1 Signature Algorithm" }
+            val authnRequest = createDefaultAuthnRequest(SamlProtocol.Binding.HTTP_REDIRECT).apply {
+                assertionConsumerServiceURL = null
+            }
+
+            val encodedRequest = encodeAuthnRequest(authnRequest)
+            // Main goal of this test is to use DSA signature algorithm to verify that the IdP
+            // can handle it. See Bindings 3.4.4.1 - DEFLATE Encoding.
+            val queryParams = SimpleSign().signUriString(
+                    SAML_REQUEST,
+                    encodedRequest,
+                    null,
+                    SSOConstants.DSA_SHA1)
+
+            // Get response from AuthnRequest
+            val response = sendRedirectAuthnRequest(queryParams)
+            BindingVerifier.verifyHttpStatusCode(response.statusCode)
+
+            val finalHttpResponse =
+                    getServiceProvider(IdpSSOResponder::class).getResponseForRedirectRequest(
+                            response)
+            val samlResponseDom = finalHttpResponse.getBindingVerifier().decodeAndVerify()
 
             // Main goal of this test is to verify the ACS in verifyAssertionConsumerService
```

Comment:
✏️ I think this comment was left over from the `"Redirect AuthnRequest Without ACS Url or ACS Index Test"` test you took inspiration from.